local/rest_api_gcbm/app.py

```diff
@@ -177,71 +183,419 @@ def gcbm_upload():
 
     # Create project directory
-    project_dir = f"{title}"
-    if not os.path.exists(f"{os.getcwd()}/input/{project_dir}"):
-        os.makedirs(f"{os.getcwd()}/input/{project_dir}")
+    input_dir = f"{os.getcwd()}/input/{title}"
+    if not os.path.exists(f"{input_dir}"):
+        os.makedirs(f"{input_dir}")
     logging.debug(os.getcwd())
 
-    # Function to flatten paths
-    def fix_path(path):
-        return os.path.basename(path.replace("\\", "/"))
-
-    # Process configuration files
-    if "config_files" in request.files:
-        for file in request.files.getlist("config_files"):
-            # Fix paths in provider_config
-            if file.filename == "provider_config.json":
-                provider_config = json.load(file)
-                provider_config["Providers"]["SQLite"]["path"] = fix_path(
-                    provider_config["Providers"]["SQLite"]["path"]
-                )
-                layers = []
-                for layer in provider_config["Providers"]["RasterTiled"]["layers"]:
-                    layer["layer_path"] = fix_path(layer["layer_path"])
-                    layers.append(layer)
-                provider_config["Providers"]["RasterTiled"]["layers"] = layers
-                with open(
-                    f"{os.getcwd()}/input/{project_dir}/provider_config.json", "w"
-                ) as pcf:
-                    json.dump(provider_config, pcf)
-            # Fix paths in modules_output
-            elif file.filename == "modules_output.json":
-                modules_output = json.load(file)
-                modules_output["Modules"]["CBMAggregatorSQLiteWriter"]["settings"][
-                    "databasename"
-                ] = "output/gcbm_output.db"
-                modules_output["Modules"]["WriteVariableGeotiff"]["settings"][
-                    "output_path"
-                ] = "output"
-                with open(
-                    f"{os.getcwd()}/input/{project_dir}/modules_output.json", "w"
-                ) as mof:
-                    json.dump(modules_output, mof)
-            else:
-                # Save file immediately
-                file.save(f"{os.getcwd()}/input/{project_dir}/{file.filename}")
+    # input files follow a strict structure
+    if not os.path.exists(f"{input_dir}/disturbances"):
+        os.makedirs(f"{input_dir}/disturbances")
+    if not os.path.exists(f"{input_dir}/classifiers"):
+        os.makedirs(f"{input_dir}/classifiers")
+    if not os.path.exists(f"{input_dir}/db"):
+        os.makedirs(f"{input_dir}/db")
+    if not os.path.exists(f"{input_dir}/miscellaneous"):
+        os.makedirs(f"{input_dir}/miscellaneous")
+
+    # store files following structure defined in curl.md
+    if "disturbances" in request.files:
+        for file in request.files.getlist("disturbances"):
+            file.save(f"{input_dir}/disturbances/{file.filename}")
     else:
         return {"error": "Missing configuration file"}, 400
```

Comment:
I think this can be changed to `return {"error": "Missing disturbances"}, 400`

Similarly, the else statements of each can return that the particular folder was not uploaded (classifiers not uploaded)

Currently, the `disturbances` folder is essential for the simulation, without which it will throw an error
To ensure it can run even without the disturbances folder maybe a `flag` can be used indicating if the disturbance folder was passed

Reply:
In the future, it is important that we can run a simulation without disturbances. Let's log this as an issue.

Reply:
We could want similar behaviour for classifiers and miscellaneous files too right?

Reply:
There will always be one classifier, and probably one misc. input, but I'm not actually sure what the minimum viable simulation is. I mention disturbances because it's a common analysis to run the same simulation with and without disturbances.